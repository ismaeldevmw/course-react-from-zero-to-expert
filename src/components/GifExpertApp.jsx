import React, { useState } from 'react'
import { AddCategory } from './AddCategory';

const GifExpertApp = () => {

    // const categories = ['One Punch', 'Samurai', 'Dragón'];
    const [categories, setCategories] = useState(['One Punch', 'Samurai', 'Dragón']);

    // const handleAdd = () => {
    //     // setCategories([...categories, 'HunterXHunter']);
    //     setCategories( cats => [...cats, 'HunterXHunter']);
    // }

    return (
        <>

            <h2>GifExpertApp</h2>
            <AddCategory />
            <hr/>

            <ol>
                {
                    categories.map( (category,  i) => {
                        return <li key={ i }>{ category }</li>
                    })
                }
            </ol>
        </>
    )
}

export default GifExpertApp
