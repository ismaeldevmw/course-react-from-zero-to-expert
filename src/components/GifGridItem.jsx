import React from 'react';
import PropTypes from 'prop-types';

export const GifGridItem = ({ id, title, url }) => {    

    return (
        <div className="card animate__animated animate__fadeIn">
            <div className="card__body">
                <img src={ url } alt={ title} />
                <p>{ title }</p>
            </div>
        </div>
    )
}

GifGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}
