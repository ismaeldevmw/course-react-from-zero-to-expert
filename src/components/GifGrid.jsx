import React, { useState, useEffect } from 'react';
import { GifGridItem } from './GifGridItem';
import { getGifs } from '../helpers/getGifs';

export const GifGrid = ({ category }) => {

    const [images, setImages] = useState([]);
    
    // It allow us to run once when the page is rendered for the first time
    useEffect( () => {
        getGifs( category )
            .then( setImages);
    }, [ category ]);
    
    return (
        <>
            <h3> { category } </h3>
            <div className="card-grid">
                {
                    images.map( img => (
                        <GifGridItem 
                            key={ img.id }
                            { ...img }
                        />
                    ))
                }          
            </div>
        </>
    )
}
