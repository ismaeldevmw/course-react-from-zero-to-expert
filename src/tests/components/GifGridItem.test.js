import React from 'react';
import { shallow } from 'enzyme';

import { GifGridItem } from '../../components/GifGridItem';


describe('Tests into <GifGridItem />', () => {
    
    test('should show <GifGridItem /> alright', () => {
        
        const wrapper = shallow(<GifGridItem />);
        expect(wrapper).toMatchSnapshot();

    });

});