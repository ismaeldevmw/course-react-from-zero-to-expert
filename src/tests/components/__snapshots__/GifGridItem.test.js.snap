// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests into <GifGridItem /> should show <GifGridItem /> alright 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <div
    className="card__body"
  >
    <img />
    <p />
  </div>
</div>
`;
